job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java

```diff
@@ -80,24 +80,44 @@ private boolean shouldIgnore(URIStatus status) {
     AlluxioURI outputDir = new AlluxioURI(config.getOutput());
 
     List<URIStatus> files = Lists.newArrayList();
+    // use double to prevent overflow
+    double totalFileSize = 0;
     for (URIStatus status : context.getFileSystem().listStatus(inputDir)) {
       if (!shouldIgnore(status)) {
         files.add(status);
+        totalFileSize += status.getLength();
```

Comment:
I have another higher level question. If there is a file conversion happening, for example, csv to parquet, there will most likely be a reduction in file sizes. This computes the INPUT file sizes, but ideally we should be computing the desired OUTPUT file sizes. Is there a simple way we can perform a heuristic for when we are converting input file types to different output types?